### ISSUE TITLE
[NEWS] 14 grudnia 2022 wprowadzimy nowe limity w zasobach do grupowej edycji ofert / On December 14, 2022, we will introduce new limits in the resources for editing offers in bulk

### ISSUE THREAD

**Issue by PrzemyslawLukanowski** [allegro/allegro-api#6434]
14 grudnia 2022 wprowadzimy nowe limity w zasobach do grupowej edycji ofert:

* 9 tys. zmian w ofertach na minutę oraz 250 tys. na godzinę przez jednego użytkownika w ramach metody PUT na zasobach:
  * [/sale/offer-price-change-commands/{commandId}](https://developer.allegro.pl/documentation/#operation/priceModificationCommandUsingPUT),
  * [/sale/offer-quantity-change-commands/{commandId}](https://developer.allegro.pl/documentation/#operation/quantityModificationCommandUsingPUT),
  * [/sale/offer-modification-commands/{commandId}](https://developer.allegro.pl/documentation/#operation/modificationCommandUsingPUT),
  * [/sale/offer-publication-commands/{commandId}](https://developer.allegro.pl/documentation/#operation/changePublicationStatusUsingPUT).
 * 270 tys. ofert w zleconych komendach na minutę w ramach metody GET na zasobach:
   * [/sale/offer-price-change-commands/{commandId}](https://developer.allegro.pl/documentation/#operation/getPriceModificationCommandStatusUsingGET),
   * [/sale/offer-price-change-commands/{commandId}/tasks](https://developer.allegro.pl/documentation/#operation/getPriceModificationCommandTasksStatusesUsingGET)
   * [/sale/offer-quantity-change-commands/{commandId}](https://developer.allegro.pl/documentation/#operation/getQuantityModificationCommandStatusUsingGET),
   * [/sale/offer-quantity-change-commands/{commandId}/tasks](https://developer.allegro.pl/documentation/#operation/getQuantityModificationCommandTasksStatusesUsingGET),
   * [/sale/offer-modification-commands/{commandId}](https://developer.allegro.pl/documentation/#operation/getGeneralReportUsingGET),
   * [/sale/offer-modification-commands/{commandId}/tasks](https://developer.allegro.pl/documentation/#operation/getTasksUsingGET),
   * [/sale/offer-publication-commands/{commandId}](https://developer.allegro.pl/documentation/#operation/getPublicationReportUsingGET),
   * [/sale/offer-publication-commands/{commandId}/tasks.](https://developer.allegro.pl/documentation/#operation/getPublicationTasksUsingGET)

**Dlaczego wprowadzamy zmianę?**

Chcemy poprawić wydajność zasobów odpowiedzialnych za przetwarzanie operacji grupowych oraz zabezpieczyć je przed nieoczekiwanym wzrostem zapytań, w tym również takich, które odbiegają od normy, np. wielokrotne zmiany ceny na tą samą wartość.  Jeśli korzystasz optymalnie z wyżej wymienionych zasobów, zmiana, którą wprowadzimy, nie będzie miała wpływu na funkcjonowanie Twojej aplikacji.

-----

On December 14, 2022, we will introduce new limits in the resources for editing offers in bulk:

* 9000 offer changes per minute and 250 000 per hour by a user within the PUT method on the following resources:
  * [/sale/offer-price-change-commands/{commandId}](https://developer.allegro.pl/documentation/#operation/priceModificationCommandUsingPUT),
  * [/sale/offer-quantity-change-commands/{commandId}](https://developer.allegro.pl/documentation/#operation/quantityModificationCommandUsingPUT),
  * [/sale/offer-modification-commands/{commandId}](https://developer.allegro.pl/documentation/#operation/modificationCommandUsingPUT),
  * [/sale/offer-publication-commands/{commandId}](https://developer.allegro.pl/documentation/#operation/changePublicationStatusUsingPUT).
* 270 000 offers in the requested commands per minute within the GET method on the following resources:
  * [/sale/offer-price-change-commands/{commandId}](https://developer.allegro.pl/documentation/#operation/getPriceModificationCommandStatusUsingGET),
  * [/sale/offer-price-change-commands/{commandId}/tasks](https://developer.allegro.pl/documentation/#operation/getPriceModificationCommandTasksStatusesUsingGET)
  * [/sale/offer-quantity-change-commands/{commandId}](https://developer.allegro.pl/documentation/#operation/getQuantityModificationCommandStatusUsingGET),
  * [/sale/offer-quantity-change-commands/{commandId}/tasks](https://developer.allegro.pl/documentation/#operation/getQuantityModificationCommandTasksStatusesUsingGET),
  * [/sale/offer-modification-commands/{commandId}](https://developer.allegro.pl/documentation/#operation/getGeneralReportUsingGET),
  * [/sale/offer-modification-commands/{commandId}/tasks](https://developer.allegro.pl/documentation/#operation/getTasksUsingGET),
  * [/sale/offer-publication-commands/{commandId}](https://developer.allegro.pl/documentation/#operation/getPublicationReportUsingGET),
  * [/sale/offer-publication-commands/{commandId}/tasks.](https://developer.allegro.pl/documentation/#operation/getPublicationTasksUsingGET)

**Why will we implement the change?**

We want to improve the efficiency of the resources responsible for processing group operations and secure them against an unexpected increase of requests, including those that differ from the norm, e.g. multiple price changes to the same value. If you are using the mentioned resources optimally, the change will not affect the functionality of your application.

**Comment by jakubmaguza** [comment]
Czym jest "użytkownik"? Użytkownikiem mojej aplikacji, czy mną (osobą, która wygenerowała ClientID)?

**Comment by PrzemyslawLukanowski** [comment]
Użytkownikiem w tym przypadku jest pojedynczy użytkownik Twojej aplikacji.

**Comment by jakubmaguza** [comment]
To już chyba nic nie rozumiem. Wcześniej pisaliście, że limit jest 9000 per ClientId.
W takim przypadku, jeśli mam 3 użytkowników, to mogę wysłać 27000 zapytań na minutę, czy 9000?

**Comment by PrzemyslawLukanowski** [comment]
Tak, limit globalny 9000 jest per clientID, jednak limity w tym wątku dotyczą zmian w ofertach oraz liczby ofert w zleconych komendach. W przytoczonym przykładzie - aplikacja nadal wysłać 9000 zapytań na minutę, a pojedynczy użytkownik może maksymalnie w tym czasie wysłać 9000 zmian - czyli jeśli w jednym requeście znajdzie się dyspozycja zmiany liczby sztuk dla 1000 ofert, to po wysłaniu 9-ciu takich żądań osiągnie limit minutowy.
